subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -0,0 +1,737 @@
+						    ext_hdr_adi_adv_data);
+
+		/* set the primary PDU len */
+		pri->len = pri_len;
+
+		/* Calc secondary PDU len */
+		_sec_len = _ps - (u8_t *)_s;
+		sec_len = ps - (u8_t *)s;
+		s->ext_hdr_len = sec_len - offsetof(struct pdu_adv_com_ext_adv,
+						    ext_hdr_adi_adv_data);
+
+		/* TODO: Check AdvData overflow */
+		ad_len = _sec->len - _sec_len;
+
+		/* set the secondary PDU len */
+		sec->len = sec_len + ad_len;
+
+		/* Fill AdvData in secondary PDU */
+		memcpy(ps, _ps, ad_len);
+
+		/* Start filling primary PDU payload based on flags */
+
+		/* No AdvData in primary channel PDU */
+
+		/* No ACAD in primary channel PDU */
+
+		/* Tx Power */
+		if (hp->tx_pwr) {
+			*--pp = *--_pp;
+		} else if (hs->tx_pwr) {
+			*--ps = *--_ps;
+		}
+
+		/* No SyncInfo in primary channel PDU */
+		/* Fill SyncInfo in secondary channel PDU */
+		ps -= sizeof(*si);
+		si = (void *)ps;
+		si->offs = 0; /* NOTE: Filled by secondary prepare */
+		si->offs_units = 0;
+		si->interval = interval;
+		memcpy(si->sca_chm, lll_sync->data_chan_map,
+		       sizeof(si->sca_chm));
+		memcpy(&si->aa, lll_sync->access_addr, sizeof(si->aa));
+		memcpy(si->crc_init, lll_sync->crc_init, sizeof(si->crc_init));
+
+		si->evt_cntr = 0; /* TODO: Implementation defined */
+
+		/* AuxPtr */
+		if (_hp.aux_ptr) {
+			_pp -= sizeof(struct ext_adv_aux_ptr);
+		}
+		{
+			struct ext_adv_aux_ptr *aux;
+
+			pp -= sizeof(struct ext_adv_aux_ptr);
+
+			/* NOTE: Aux Offset will be set in advertiser LLL event
+			 */
+			aux = (void *)pp;
+			aux->chan_idx = 0; /* FIXME: implementation defined */
+			aux->ca = 0; /* FIXME: implementation defined */
+			aux->offs_units = 0; /* FIXME: implementation defined */
+			aux->phy = find_lsb_set(adv->lll.phy_s) - 1;
+		}
+
+		/* TODO: reduce duplicate code if below remains similar to
+		 * primary PDU
+		 */
+		if (_hs.aux_ptr) {
+			struct ext_adv_aux_ptr *aux;
+
+			_ps -= sizeof(struct ext_adv_aux_ptr);
+			ps -= sizeof(struct ext_adv_aux_ptr);
+
+			/* NOTE: Aux Offset will be set in advertiser LLL event
+			 */
+			aux = (void *)ps;
+			aux->chan_idx = 0; /* FIXME: implementation defined */
+			aux->ca = 0; /* FIXME: implementation defined */
+			aux->offs_units = 0; /* FIXME: implementation defined */
+			aux->phy = find_lsb_set(adv->lll.phy_s) - 1;
+		}
+
+		/* ADI */
+		{
+			struct ext_adv_adi *ap, *as;
+			u16_t did = UINT16_MAX;
+
+			pp -= sizeof(struct ext_adv_adi);
+			ps -= sizeof(struct ext_adv_adi);
+
+			ap = (void *)pp;
+			as = (void *)ps;
+
+			if (_hp.adi) {
+				struct ext_adv_adi *_adi;
+
+				_pp -= sizeof(struct ext_adv_adi);
+				_ps -= sizeof(struct ext_adv_adi);
+
+				/* NOTE: memcpy shall handle overlapping buffers
+				 */
+				memcpy(pp, _pp, sizeof(struct ext_adv_adi));
+				memcpy(ps, _ps, sizeof(struct ext_adv_adi));
+
+				_adi = (void *)_pp;
+				did = _adi->did;
+			} else {
+				ap->sid = adv->sid;
+				as->sid = adv->sid;
+			}
+
+			did++;
+
+			ap->did = did;
```

Comment:
As far as I can tell, this should be okay without endianness conversion. However, it would be preferable to convert endianness anyway to be able to have identical traces between LE and BE architectures.

Reply:
good catch. Fixed.